spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/convert/DefaultDbRefResolver.java

```diff
@@ -106,7 +107,7 @@ public DBRef createDbRef(org.springframework.data.mongodb.core.mapping.DBRef ann
 	 */
 	@Override
 	public DBObject fetch(DBRef dbRef) {
-		return ReflectiveDBRefResolver.fetch(mongoDbFactory.getDb(), dbRef);
+		return ReflectiveDBRefResolver.fetch(MongoClientVersion.isMongo3Driver() ? mongoDbFactory.getDb() : null, dbRef);
```

Comment:
the `ReflectiveDbRefResolver` is intended to be a only short living construct (should be removed once we fully switch to MongoDB Java driver 3.x). Therefore I wanted to explicitly check the `DbRefResolver.  We'll need to have further checks for DATAMONGO-1194 there as well.
But sure we can also do it the other way - just let me know.

Since we need to have a look at the DB handling in DATAMONGO-1194 anyway I also tried to keep the changes modest.